thirdparty/faiss/faiss/IndexIVF.cpp

```diff
@@ -933,6 +934,182 @@ void IndexIVF::range_search_preassigned(
     }
 }
 
+std::unique_ptr<IVFIteratorWorkspace> 
+IndexIVF::getIteratorWorkspace(
+        const float* query_data,
+        const IVFSearchParameters* ivfsearchParams) const {
+    return std::make_unique<IVFIteratorWorkspace>(
+            query_data, ivfsearchParams);
+}
+
+std::optional<std::pair<float, idx_t>> IndexIVF::getIteratorNext(
+        IVFIteratorWorkspace* workspace) const {
+    auto scan_one_list_then_add_to_backup =
+            [&](idx_t list_no,
+                float coarse_list_centroid_dist, // no use, dist for residual.
+                float* distances,
+                idx_t* labels,
+                size_t& counter_back,
+                size_t max_codes) {
+                if (list_no < 0) {
+                    // not enough centroids for multiprobe
+                    return (size_t)0;
+                }
+                FAISS_THROW_IF_NOT_FMT(
+                        list_no < (idx_t)nlist,
+                        "Invalid list_no=%" PRId64 " nlist=%zd\n",
+                        list_no,
+                        nlist);
+
+                // don't waste time on empty lists
+                if (invlists->is_empty(list_no)) {
+                    return (size_t)0;
+                }
+
+                // get scanner
+                IDSelector* sel = workspace->search_params
+                        ? workspace->search_params->sel
+                        : nullptr;
+                InvertedListScanner* scanner = get_InvertedListScanner(false, sel);
+                scanner->set_query(workspace->query_data);
+                scanner->set_list(list_no, coarse_list_centroid_dist);
+                ScopeDeleter1<InvertedListScanner> del(scanner);
+
+                size_t segment_num = invlists->get_segment_num(list_no);
+                size_t scan_cnt = 0;
+                for (size_t segment_idx = 0; segment_idx < segment_num;
+                     segment_idx++) {
+                    size_t segment_size =
+                            invlists->get_segment_size(list_no, segment_idx);
+                    size_t should_scan_size =
+                            std::min(segment_size, max_codes - scan_cnt);
+                    scan_cnt += should_scan_size;
+                    if (should_scan_size <= 0) {
+                        break;
+                    }
+                    size_t segment_offset =
+                            invlists->get_segment_offset(list_no, segment_idx);
+                    InvertedLists::ScopedCodes scodes(
+                            invlists, list_no, segment_offset);
+                    InvertedLists::ScopedCodeNorms scode_norms(
+                            invlists, list_no, segment_offset);
+                    InvertedLists::ScopedIds sids(
+                            invlists, list_no, segment_offset);
+
+                    size_t n_heap = scanner->scan_codes_and_push_back(
```

Comment:
temp variable n_heap is not used. discussed with @alwayslove2013 and we can remove it.

Reply:
sorry for being not precise in the previous comment: I meant to simply remove this local variable while `scan_codes_and_push_back` can still keep the return value for future use(could be used for metrics) and to have a aligned behavior with `scan_codes`.